apps/src/templates/policy_compliance/LockoutLinkedAccounts.jsx

```diff
@@ -0,0 +1,311 @@
+    return isEmail(email) && props.userEmail !== hashString(email);
+  };
+
+  // Set the disabled state of the submit button based on the validity of the
+  // email in the field.
+  const [disabled, setDisabled] = useState(
+    () => !validateEmail(props.pendingEmail)
+  );
+
+  // When the email field is updated, also update the disability state of the
+  // submit button.
+  const onEmailUpdate = event => {
+    setDisabled(!validateEmail(event.target.value));
+  };
+
+  // This will set the email to the current pending email and fire off the
+  // form as though they had typed in the same email again.
+  const resendPermissionEmail = event => {
+    const field = document.getElementById('parent-email');
+    field.value = props.pendingEmail;
+
+    const form = document.getElementById('lockout-linked-accounts-form');
+    form.submit();
+  };
+
+  // Get the current locale.
+  const locale = cookies.get('language_') || 'en-US';
+
+  // Whether or not we are rendering right-to-left.
+  const isRTL = getStore().getState()?.isRtl;
+
+  // How to format any localized dates on the page.
+  const dateOptions = {
+    year: 'numeric',
+    month: 'long',
+    day: 'numeric',
+  };
+
+  // We want to mark the email in the prompt in bold, so we split the
+  // localized string. It is a slight hack since we cannot just embed React
+  // component code into a string. Should work as though we could though.
+  const pendingPrompt = i18n.sessionLockoutPendingPrompt({
+    pendingEmail: '{pendingEmail}',
+  });
+  const pendingPromptParts = pendingPrompt.split('{pendingEmail}');
+
+  const tokenElement = document.querySelector('meta[name="csrf-token"]');
+  let csrfToken = '';
+  if (tokenElement) {
+    csrfToken = tokenElement.attributes['content'].value;
+  }
+
+  // Child permission status from the user record
+  const permissionStatus = {};
+  switch (props.permissionStatus) {
+    case ChildAccountComplianceStates.REQUEST_SENT:
+      permissionStatus.message = i18n.sessionLockoutStatusPending();
+      permissionStatus.style = styles.pending;
+      break;
+    case ChildAccountComplianceStates.PERMISSION_GRANTED:
+      permissionStatus.message = i18n.sessionLockoutStatusGranted();
+      permissionStatus.style = styles.granted;
+      break;
+    default:
+      permissionStatus.message = i18n.sessionLockoutStatusNotSubmitted();
+      permissionStatus.style = styles.notSubmitted;
+  }
+
+  // Custom form handler to submit the permission request. The default form
+  // submission does not include the Referer header, which is required for the
+  // policy_compliance_controller to redirect back to the correct page.
+  const submitPermissionRequest = async e => {
+    e.preventDefault();
+    const params = new URLSearchParams({
+      'parent-email': e.target['parent-email'].value,
+      authenticity_token: csrfToken,
+    });
+
+    await fetch(props.apiUrl, {
+      method: 'POST',
+      body: params,
+    });
+    window.location.reload();
```

Comment:
Ah, the `fetch` sends the `HTTP_REFERER` but not a normal `submit()`? I remember this issue. Well, one thing you could do is actually just handle this as part of the component. Do something like a spinner while the fetch happens. After the fetch, see if the response is successful and just set the state of the panel. Then it will just re-render itself.

Reply:
Ah, I see we do have a spinner component - I will look into this!

Reply:
@wilkie The spinner was a good suggestion. The state change refreshes the react component, but since the important data for the request is set in the controller, refreshing just the one component doesn't seem to do the trick. I've added the spinner to communicate the loading feel to the user, but still have the page reload in there. Is there a better way to do this with React, to selectively retrieve fresh backend data?

Reply:
When the POST returns, that state should exist. The POST route in the controller could return it in a JSON block or you can just interpret a 200 from the POST as meaning that the request succeeded and you update the permission status of the component. Alternatively, adding a route to query the status of the lockout request... possibly just reacting to asking for `/lockout` with `application-json` so it returns JSON instead of HTML. That's maybe the most extreme.

Let's say the POST just returned the state information you need. That would still mean that you would need a permission status state that you initially set to the incoming status property. Then set that state based on the information you are getting back from the POST request.

I suppose a reload isn't really that bad, though. :) What works, works.